lib/ConsumerImpl.cc

```diff
@@ -319,6 +320,31 @@ void ConsumerImpl::unsubscribeAsync(ResultCallback originalCallback) {
     }
 }
 
+void ConsumerImpl::triggerCheckExpiredChunkedTimer() {
+    checkExpiredChunkedTimer_->expires_from_now(
+        boost::posix_time::milliseconds(expireTimeOfIncompleteChunkedMessageMs_));
+    std::weak_ptr<ConsumerImplBase> weakSelf{shared_from_this()};
+    checkExpiredChunkedTimer_->async_wait([this, weakSelf](const boost::system::error_code& ec) -> void {
```

Comment:
Here still caught `this`, You can refer:

``` c++
    std::weak_ptr<ConsumerImpl> weakSelf{get_shared_this_ptr()};
    checkExpiredChunkedTimer_->async_wait([weakSelf](const boost::system::error_code& ec) -> void {
        auto self = weakSelf.lock();
        if (!self || ec) {
            return;
        }
        Lock lock(self->chunkProcessMutex_);
        long currentTimeMs = TimeUtils::currentTimeMillis();
        self->chunkedMessageCache_.removeOldestValuesIf(
            [self, &currentTimeMs](const std::string& uuid, const ChunkedMessageCtx& ctx) -> bool {
                return currentTimeMs > ctx.getReceivedTimeMs() + self->expireTimeOfIncompleteChunkedMessageMs_;
            });
        self->triggerCheckExpiredChunkedTimer();
        return;
    });

```

Reply:
I think it's okay for the current approach.

Reply:
Make sense.